core/platform/auth/gae_auth_services_test.py

```diff
@@ -33,6 +33,31 @@ class GaeAuthServicesTests(test_utils.GenericTestBase):
 
     ENABLE_AUTH_SERVICES_STUB = False
 
+    def test_establish_auth_session_does_nothing(self):
+        request = webapp2.Request.blank('/')
+        response = webapp2.Response()
+        # Does not raise.
+        gae_auth_services.establish_auth_session(request, response)
+
+    def test_destroy_auth_session_deletes_cookies(self):
+        response = webapp2.Response()
+        response.set_cookie('ACSID', value='abc')
+        response.set_cookie('SACSID', value='def')
+        response.set_cookie('dev_appserver_login', value='ghi')
+        response.set_cookie('custom', value='123')
+
+        gae_auth_services.destroy_auth_session(response)
+
+        self.assert_matches_regexps(response.headers.get_all('Set-Cookie'), [
```

Comment:
I don't quite understand what's going on here. The old cookies are still present -- that's not so good, right?

Was that the case before, with the old code?

Reply:
OK, have fixed it. Order matters in HTTP headers, and for cookies the last value takes precedence. Have improved the regexps to reflect this.

Reply:
Same question still though. Was this the case before with the old code? If the old code actually removes the old stuff then I would prefer it.

Reply:
Yes it was, the old code did regexp to replace the Max-Age with a date in the past.

Reply:
But I think that is better, to remove the old thing altogether. If it wasn't broken before, I wouldn't change the impl.